woeip/apps/air_quality/views.py

```diff
@@ -1,61 +1,40 @@
 import logging
 
 from django.contrib import messages
-from django.contrib.auth.decorators import login_required
-from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import redirect, render
-from django.utils.encoding import force_text
+from django.views import View
 
-from woeip.apps.air_quality import dustrak, forms, models
+from .forms import SessionDataForm
+from .models import SessionData
 
 logger = logging.getLogger(__name__)
 
 
-@login_required
-def upload(request):
-    """Upload data for a session collected using the Dustrak air quality device and a separate GPS
-    log file.
-    """
-    request_user = request.user
-    if request.method == 'POST':
-        form = forms.DustrakSessionForm(request.POST, request.FILES)
-        form_instance = form.instance
+class Upload(LoginRequiredMixin, View):
+    def get(self, request):
+        """Present file upload form to user"""
+        return render(self.request, 'air_quality/upload.html', {
+            'form': SessionDataForm
+        })
+
+    def post(self, request):
+        """Save files to SessionData table"""
+        files = self.request.FILES
+        form = SessionDataForm(self.request.POST, files)
```

Comment:
A more standard syntax for this section would be (esp. better to avoid the circular redirect)
```
    if form.is_valid():
        form.save()
        return redirect('view_session_data')
    else:
        messages.add_message(self.request, messages.ERROR, 'File upload error')
return return render(self.request, 'air_quality/upload.html', {
    'form': SessionDataForm
})

Reply:
- I like your suggestion to render the upload page again, rather than redirecting. I'll incorporate that change.
- As far listing return twice, that's how I originally structured it. Unfortunately, it causes a linting error. This raises a dilemma between suppressing linting warnings or going against django convention. I ended choosing the later for two reasons. 
 1. I think suppressing warnings is a dangerous game
 2. My freshman python instructor taught me that functions should only have one return. Django be damned, I'm  following my instructor!

Anyway, I'd like to keep the 'path' style with a single return. But, if you're adamant about two returns, then I'll change it.

Reply:
I've heard of single return style for other languages but it isn't commonly used for Python - neither PEP8 nor Google style guide prescribe it. Since Python does automatic garbage collection, there isn't a programmatic downside to doing multiple returns, except in cases where it makes the code easier to read not to do them. In this case, I think having multiple returns makes the code easier to read since as you mentioned, multiple returns are idiomatic to django (e.g. in the [docs for class-based views](https://docs.djangoproject.com/en/2.2/topics/class-based-views/intro/#handling-forms-with-class-based-views)), and it's usual syntax in django views to return the name of the view in the return statement.

One of the linters we're using (pylint) is currently set to the default settings (in the `pylintrc` file). This sometimes doesn't work well for a specific use case like web development (e.g. another example is that there is a rule against too many ancestors, which doesn't work well for django, which nests a lot of imports). It's pretty common to configure specific linter settings per codebase, and I think it's quite reasonable for us to modify these settings if it's sensible.

Btw I realized I had an extraneous `else` in my suggestion, so here's an even simpler version:

```
    if form.is_valid():
        form.save()
        return redirect('view_session_data')
messages.add_message(self.request, messages.ERROR, 'File upload error')
return render(self.request, 'air_quality/upload.html', {
    'form': SessionDataForm
})
```

If you still prefer the single return syntax, I would suggest using a variable name like `view_result` or something like that since `path` is a bit ambiguous and could be easily confused with `os.path` and `sys.path`.

Reply:
I'm convinced. I'll have it changed by this Hack Night.

I also think the linter configuration deserves its own issue.

Reply:
> I'm convinced. I'll have it changed by this Hack Night.
> 
> I also think the linter configuration deserves its own issue.

SGTM for both!